backend/routes/products.js

```diff
@@ -34,4 +34,36 @@ router.post("/add-new", async (req, res) => {
   res.status(500).send("Product creation failed. please try again later");
 });
 
+router.put("/:id", async (req, res) => {
+  const productId = req.params.id;
+  let params = {
+    title: req.body.title,
+    image: req.body.image,
+    description: req.body.description,
+    category: req.body.category,
+    price: req.body.price,
+  };
+  for (let prop in params) if (!params[prop]) delete params[prop]; //Goes through the params and deletes them if they have a falsy value.
```

Comment:
check for productId before anything if you whish to handle faulty requests.
first check productId, then continue to do all the object mapping..